library/arith.lisp

```diff
@@ -480,6 +493,136 @@ The fields are defined as follows:
       (lisp Double-Float (x y)
         (cl:coerce (cl:/ x y) 'cl:double-float)))))
 
+(coalton-toplevel
+  (define-class (Trigonometric :a)
+    (sin   (:a -> :a))
+    (cos   (:a -> :a))
+    (tan   (:a -> :a))
+    (sinh  (:a -> :a))
+    (cosh  (:a -> :a))
+    (tanh  (:a -> :a))
+    (asin  (:a -> :a))
+    (acos  (:a -> :a))
+    (atan  (:a -> :a))
+    (asinh (:a -> :a))
+    (acosh (:a -> :a))
+    (atanh (:a -> :a)))
+
+  (declare sincos (Trigonometric :a => :a -> (Tuple :a :a)))
+  (define (sincos x)
+    (tuple (sin x) (cos x)))
+
+  (define-class (Exponentiable :a)
+    (**      (:a -> :a -> :a))
```

Comment:
you may want to re-use Common Lisp's names `expt` and `log`

Reply:
only objection is that in Lisp `log` is value then base, not base then value.

I much prefer base then value, as you have

Reply:
i'm ashamed to admit i don't remember the difference between ^ ^^ **

Reply:
- `^ x n` multiplies any `Num` `x` by itself `n` times (`n` is an integer)
- `^^ x n` multiplies or divides any `Dividable` `x` by itself `n` times (`n` is an integer)
- `** x y` exponentiates any algebriac/transcendental numbers `x` and `y`
- `expt` (in common lisp) does all of the above